js/index.js

```diff
@@ -9,14 +9,56 @@ var UI = require('./ui.js'),
 function bindToForm(form) {
 	form.addEventListener('change', UI.reflectParameterChange)
 
-	var update = debounce(OpenFisca.update.bind(form), 300)
+	var handleBasicFormChanges = debounce(OpenFisca.update.bind(form), 300)
 
-	update()
+	handleBasicFormChanges()
 
-	form.addEventListener('change', update)
-	form.addEventListener('keyup', update)
+	var handleFormChanges = function(event) {
+		handlePostalCodeInput(event, handleBasicFormChanges)
+	}
+
+	form.addEventListener('change', handleFormChanges)
+	form.addEventListener('keyup', handleFormChanges)
 }
 
+/*
+code_postal_entreprise <input> value is used to dynamically
+fetch a list of options for the the depcom_entreprise <select> element
+*/
+function handlePostalCodeInput(event, next) {
+
+	if (event.target.name !== 'code_postal_entreprise')
```

Comment:
Move this precondition evaluation to the caller: the function is called `handlePostalCodeInput`, it should be called when handling a postal code input.

> This will help scale the `handleFormChanges` dispatcher when other handlers are added.